flink-batch-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCInputFormat.java

```diff
@@ -363,14 +439,31 @@ public JDBCInputFormat finish() {
 			if (format.dbURL == null) {
 				throw new IllegalArgumentException("No database URL supplied");
 			}
-			if (format.query == null) {
+			if (format.queryTemplate == null) {
 				throw new IllegalArgumentException("No query supplied");
 			}
 			if (format.drivername == null) {
 				throw new IllegalArgumentException("No driver supplied");
 			}
+			adjustQueryTemplateIfNecessary();
 			return format;
 		}
+
+		/** Try to add $CONDITIONS token automatically (at least for straightforward cases) */
+		private void adjustQueryTemplateIfNecessary() {
```

Comment:
Indeed I only try to fix the query if it is a simple select (I add the WHERE clause). Otherwise it fails (I agree that is very dangerous to try to guess where to put the $CONDITIONS token).